farm/modeling/language_model.py

```diff
@@ -290,7 +294,22 @@ def save(self, save_dir):
         model_to_save = (
             self.model.module if hasattr(self.model, "module") else self.model
         )  # Only save the model it-self
-        torch.save(model_to_save.state_dict(), save_name)
+
+        state_dict = model_to_save.state_dict()
+        keys = state_dict.keys()
+
+        # If we save a question_encoder or a ctx_encoder of a dpr model that is not based on a standard BERT model,
+        # e.g., a Camembert model, we need to adjust the names of the model weights.
+        # This adjustment removes a prefix name that otherwise would prevent the weights from being loaded again.
+        if self.model.config.model_type !="dpr" and (model_to_save.base_model_prefix.startswith("question_") or model_to_save.base_model_prefix.startswith("ctx_")):
```

Comment:
Could we not move this part to the DPRQuestion- and ContextEncoder class by adding a save method there and calling the superclass' save method with altered state dicts? Should work, as mentioned here: https://discuss.pytorch.org/t/how-to-change-the-name-of-the-weights-to-a-new-name-when-saving-the-model/32166/2